src/apprt/gtk/Tab.zig

```diff
@@ -121,10 +121,71 @@ pub fn remove(self: *Tab) void {
     self.window.closeTab(self);
 }
 
+/// Helper function to check if any surface in the split hierarchy needs close confirmation
+const needsConfirm = struct {
+    fn check(elem: Surface.Container.Elem) bool {
+        return switch (elem) {
+            .surface => |s| s.core_surface.needsConfirmQuit(),
+            .split => |s| check(s.top_left) or check(s.bottom_right),
+        };
+    }
+}.check;
```

Comment:
I'm aware of this pattern but its not something we generally do within this project.

I don't think we need to wrap this in a struct at all and can just make it a top-level (non-pub) `fn`.